packages/rs-drive/src/drive/identity/fetch/partial_identity/fetch_identity_balance_with_keys/v0/mod.rs

```diff
@@ -78,9 +78,12 @@ impl Drive {
         transaction: TransactionArg,
         platform_version: &PlatformVersion,
     ) -> Result<(Option<PartialIdentity>, FeeResult), Error> {
-        let balance_cost = epoch.cost_for_known_cost_item(FetchIdentityBalanceProcessingCost);
+        let cached_fee_version = self.cache.cached_fee_version.read();
```

Comment:
Why do we read it multiple times?

Reply:
Well, read() acquires read lock everytime we want to access the cached map. I don't if we can do it once on an upper level :man-shrugging: